examples/comet.py

```diff
@@ -0,0 +1,86 @@
+import taichi as ti
+import taichi_glsl as tl
+ti.init(arch=[ti.cuda, ti.metal])
+
+dim = 3
+N = 1024 * 8
+dt = 2e-4
+steps = 7
+sun = ti.Vector([0.5, 0.5] if dim == 2 else [0.5, 0.5, 0.0])
+gravity = 0.5
+pressure = 0.3
+pps_scale = 0.4
+colorinit = 0.3
+colordeg = 1.6
+initvel = 0.07
+res = 640
+
+inv_m = ti.field(ti.f32)
+color = ti.field(ti.f32)
+x = ti.Vector.field(dim, ti.f32)
+v = ti.Vector.field(dim, ti.f32)
+ti.root.bitmasked(ti.i, N).place(x, v, inv_m, color)
+xcnt = ti.field(ti.i32, ())
+img = ti.field(ti.f32, (res, res))
+
+
+@ti.kernel
+def substep():
+    ti.no_activate(x)
+    for i in x:
```

Comment:
Can we automatically use `ti.no_activate(x)` when:
1. the following struct-for is on `x`.
2. within that struct-for, only `x[i]` is written.

e.g.:
```py
for i in x:
  x[i] = 1  # OK to optimize
  x[i // 2] = 1  # won't optimize
```

Thoughts? @xumingkuan @yuanming-hu